File: app/src/main/java/tech/diarmaid/koohiiaite/fragment/KanjiDetailFragment.java
package tech.diarmaid.koohiiaite.fragment;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import tech.diarmaid.koohiiaite.R;
import tech.diarmaid.koohiiaite.adapter.KanjiDetailAdapter;

/**
 * Allows next and previous navigation by swapping this fragment
 *
 * Replaced by new versions of itself when Next/Prev pressed
 */
public class KanjiDetailFragment extends Fragment {

    private ViewPager vPager;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        //fragment instance is retained across Activity re-creation (device rotation)
        setRetainInstance(true); //may cause memory leaks according to stackoverflow
        Bundle arguments = getArguments();
        AppCompatActivity parent = (AppCompatActivity) getActivity();
        View view = inflater.inflate(R.layout.fragment_detail, container, false);
        vPager = (ViewPager) view.findViewById(R.id.vpPager);
        FragmentPagerAdapter adapterViewPager = new KanjiDetailAdapter(getChildFragmentManager(), arguments, parent);
        vPager.setAdapter(adapterViewPager);
        vPager.setCurrentItem(arguments.getInt("currentPage")); //preserve page between next/prev operations
        return view;
    }

    public int getCurrentPagerIndex()
    {
        return vPager == null ? -1 : vPager.getCurrentItem();
    }
}

File: app/src/main/java/tech/diarmaid/koohiiaite/fragment/KanjiDetailFragment.kt
package tech.diarmaid.koohiiaite.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import tech.diarmaid.koohiiaite.adapter.KanjiDetailAdapter
import tech.diarmaid.koohiiaite.databinding.FragmentDetailBinding
import tech.diarmaid.koohiiaite.viewmodel.KanjiDetailViewModel

/**
 * Allows next and previous navigation by swapping this fragment
 *
 * Replaced by new versions of itself when Next/Prev pressed
 */
class KanjiDetailFragment(val heisigId: Int) : Fragment() {

    private var _binding: FragmentDetailBinding? = null
    private val binding get() = _binding!!
    var currentPagerIndex: Int = 0
    private var adapterViewPager : KanjiDetailAdapter? = null
    private var viewModel: KanjiDetailViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProvider(this).get(KanjiDetailViewModel::class.java)
        viewModel?.heisigId?.postValue(heisigId)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        //fragment instance is retained across Activity re-creation (device rotation)
        retainInstance = true //may cause memory leaks according to stackoverflow
        val parent = activity as AppCompatActivity?
        _binding = FragmentDetailBinding.inflate(inflater, container, false)
        adapterViewPager = KanjiDetailAdapter(childFragmentManager, arguments!!, parent!!, viewModel, viewLifecycleOwner)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        currentPagerIndex = binding.vpPager.currentItem
        binding.vpPager.adapter = adapterViewPager
        binding.vpPager.currentItem =
            arguments?.getInt("currentPage") ?: 0  //preserve page between next/prev operations
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
